### PR TITLE
[SYCL] Support __builtin_printf for SYCL device

### PR DESCRIPTION
**Pull request by srividya-sundaram**

### CODE REVIEW

**Comment by premanandrao** [comment]
Could you also a CodeGen test for this to check if we do the right thing after we allow this through in Sema?

**Comment by smanna12** [comment]
> Could you also a CodeGen test for this to check if we do the right thing after we allow this through in Sema?

I agree with Prem.

File: clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -emit-llvm %s -o - | FileCheck %s
+// This test checks if __builtin_printf does not throw an error when
```
Comment (elizabethandrews) [inline clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp:2]: Please update the comment. This test checks IR, not diagnostics

**Review by elizabethandrews — APPROVED** [review]

**Comment by srividya-sundaram** [comment]
@intel/llvm-gatekeepers Hi, this is ready for merging. Thanks!

**Comment by steffenlarsen** [comment]
Failures in CI are seen on other PRs (for example https://github.com/intel/llvm/pull/7563) and have been reported in https://github.com/intel/llvm/pull/6524#issuecomment-1330376519.

**Comment by pvchupin** [comment]
@srividya-sundaram, can you please follow up on Windows failure in post-commit for the new test you've added: https://github.com/intel/llvm/actions/runs/3576029259/jobs/6013328053

**Comment by srividya-sundaram** [comment]
> @srividya-sundaram, can you please follow up on Windows failure in post-commit for the new test you've added: https://github.com/intel/llvm/actions/runs/3576029259/jobs/6013328053

Hi! Is there a way I can reproduce the post-commit failure locally on a Windows machine?
I do have a fix looking at the log file, but I want to make sure it does really fix the failure.

**Comment by premanandrao** [comment]
> Is there a way I can reproduce the post-commit failure locally on a Windows machine?

I think you should be able to run the command-line on a Windows local build.

**Comment by srividya-sundaram** [comment]
> @srividya-sundaram, can you please follow up on Windows failure in post-commit for the new test you've added: https://github.com/intel/llvm/actions/runs/3576029259/jobs/6013328053

@pvchupin Just curious. Would it have been possible to capture this failure before it got merged? 
I am not aware of the details of the post-commit checks, but wondering if it would help to have such failures be captured sooner than later in the pipeline.

**Comment by premanandrao** [comment]
> Would it have been possible to capture this failure before it got merged? I am not aware of the details of the post-commit checks, but wondering if it would help to have such failures be captured sooner than later in the pipeline.

Does it not show up in a check-clang test run on Windows?

**Comment by pvchupin** [comment]
> @pvchupin Just curious. Would it have been possible to capture this failure before it got merged? I am not aware of the details of the post-commit checks, but wondering if it would help to have such failures be captured sooner than later in the pipeline.

We don't run windows in pre-commit (due to longer pre-commit time). So you could detect this only if run locally.
It's ok to address windows specific issues post-commit.

**Comment by elizabethandrews** [comment]
> > Would it have been possible to capture this failure before it got merged? I am not aware of the details of the post-commit checks, but wondering if it would help to have such failures be captured sooner than later in the pipeline.
> 
> Does it not show up in a check-clang test run on Windows?

Yes it would but I very rarely test my patch on Windows because I work on Linux and I suspect its the same for most developers who work on Linux. I don't think we're expected to do this anyway.

**Comment by premanandrao** [comment]
> Yes it would but I very rarely test my patch on Windows because I work on Linux and I suspect its the same for most developers who work on Linux. I don't think we're expected to do this anyway.

No, I understand.  I am that way too.   I was worried initially that local runs on Windows were not showing the failure; but then I understood the question was more about the expected development/testing process.

**Pull request merged**